### PR TITLE
Clean dist folder of files before running babel

### PR DESCRIPTION
**Pull request by melonmanchan**
This fixes the issue found in #173 by cleaning the dist folder before prepublishing using the 'del' module, as recommended by: https://github.com/gulpjs/gulp/blob/master/docs/recipes/delete-files-folder.md

### CODE REVIEW

**Comment by Swiip** [comment]
`del` is an asynchronous operation. You should wait for del to finish before launching Babel. Even put del in another task.

**Comment by zckrs** [comment]
And `del@2.02` returns a promise.

```
gulp.task('clean', function () {
 return del('dist');
});
```

**Comment by melonmanchan** [comment]
Thanks for the input. How's it look now?

**Comment by zckrs** [comment]
:+1:

**Comment by melonmanchan** [comment]
SIlly me, looks like I forgot to update the tests. Will fix.

      AssertionError: gulpfile.js did not match 'gulp.task('prepublish', ['nsp', 'babel']);'. Contained:

File: generators/gulp/templates/gulpfile.js [hunk 1]
```diff
@@ -76,6 +77,10 @@ gulp.task('babel', function () {
     .pipe(babel())
     .pipe(gulp.dest('dist'));
 });
+
+gulp.task('clean', function() {
+  return del('dist/**/*');
```
Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: Doesn't `del('dist')` work?

**Comment by melonmanchan** [comment]
Yeah, del('dist') also works. Additionally, there exists a race condition between build and clean at the moment, as the dependancy tasks of build are run async, which might not be immediately apparent from Gulp's current syntax. Adding 'clean' as a dependancy should be the way to go, or at least until Gulp 4 is released, which promises to fix that particular pitfall.

**Comment by SBoudrias** [comment]
Yes, that'd effectively be better!

**Comment by melonmanchan** [comment]
Is this ready to be merged, or is there still something?

File: generators/gulp/index.js [hunk 2]
```diff
@@ -69,6 +70,7 @@ module.exports = generators.Base.extend({
       }
 
       if (this.options.babel) {
+        prepublishTasks.push('clean');
```
Comment (SBoudrias) [inline generators/gulp/index.js]: No need to have it here anymore as it's a dependency the babel task.

File: generators/gulp/templates/gulpfile.js (hunk 1 above)
Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: If as you said this syntax work, please update :)

**Comment by melonmanchan** [comment]
Fair enough! Removed the unnecessary dependancy from prepublish task and used the suggested way of cleaning up dist with del('dist). By the way, if the lib folder is totally empty dist will simply be removed and an empty folder won't be created. Is this functioning as intended?

**Comment by hemanth** [comment]
:+1:

File: generators/gulp/templates/gulpfile.js [hunk 3]
```diff
@@ -71,11 +72,15 @@ gulp.task('coveralls', ['test'], function () {
...
 <% if (babel) { -%>
 
-gulp.task('babel', function () {
+gulp.task('babel', ['clean'], function () {
   return gulp.src('<%- projectRoot %>')
     .pipe(babel())
     .pipe(gulp.dest('dist'));
 });
+
+gulp.task('clean', function() {
```
Comment (arthurvr) [inline generators/gulp/templates/gulpfile.js]: `function () {` (spacing)

**Comment by SBoudrias** [comment]
Thanks!

**Pull request merged**